cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -899,45 +899,76 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
             let lhs = input_to_reg_mem(ctx, inputs[0]);
             let rhs = input_to_reg(ctx, inputs[1]);
             let dst = output_to_reg(ctx, outputs[0]);
+            let ty = ty.unwrap();
+
+            // Move the value to the correct register. TODO Why? This seems unnecessary and potentially incorrect--at best it should be explained here.
```

Comment:
This is because the `dst` operand in `xmm_rm_r` is the read-write operand, so we need to generate a move here, at the virtual register level. It doesn't necessarily materialize as a move instruction, because regalloc will try hard to coalesce the two virtual registers (and assign them the same physical register).